pre_commit_dbt/tracking.py

```diff
@@ -0,0 +1,93 @@
+import os
+from typing import Any
+from typing import Dict
+from typing import NoReturn
+from typing import Optional
+
+from mixpanel import Mixpanel
+
+from pre_commit_dbt.utils import get_config_file
+
+MIXPANEL_DEV_ENV = "34ffa16dc37f248c18ad6d1b9ea9c3a8"
+MIXPANEL_PROD_ENV = "3fa3db873f6950d10bd770a49c57e33e"
+
+
+class dbtCheckpointTracking:
+    def __init__(self, script_args: Dict[str, Any]):
+        config_path = script_args.get("config")
+        if config_path is None or not isinstance(config_path, str):
+            raise ValueError("config_path must be a non-empty string")
+
+        self.config = get_config_file(config_path)
+
+        self.script_args = script_args
+        self.token = self._get_mixpanel_env_token()
+        self.disable_tracking = self.config.get("disable-tracking", False)
+
+    def track_hook_event(
+        self,
+        event_name: str,
+        event_properties: Dict[str, Any],
+        manifest: Dict[str, Any],
+    ) -> NoReturn:
+        if not self.disable_tracking:
+            dbt_metadata = manifest.get("metadata")
```

Comment:
Change dbt_metadata to the following:
`manifest.get("metadata", {})`

If metadata does not exist, it will return an empty dictionary instead of None

Reply:
Thats odd because there is a `try except` logic above to return an error if the `manifest` file is not detected